### PR TITLE
Update transformer examples with built in MultiHeadAttention layer

### PR DESCRIPTION
**Pull request by apoorvnandan**
Changed two examples:
- Text classification with transformer
- Text generation with miniGPT

I have tested both the changes on Colab. 
Once the review is done, I can either make the same changes in the `md` and `ipynb` files, or regenerate them. (Regenerating takes a lot of time for the mini GPT example and the output pretty much remains the same)

### CODE REVIEW

File: examples/generative/text_generation_with_miniature_gpt.py [hunk 1]
```diff
@@ -39,99 +39,31 @@
...
-        return output
-
 
 """
 ## Implement a Transformer block as a layer
 """
 
 
+def causal_attention_mask(batch_size, n_dest, n_src, dtype):
+    """
```
Comment (fchollet) [inline examples/generative/text_generation_with_miniature_gpt.py:49]: Please add a one-line description of the purpose of the causal mask.

**Review by fchollet — COMMENTED** [review]
Looks good, thank you for the PR! Please add the generated files.

**Comment by apoorvnandan** [comment]
Added the generated files. 
I made 3 hyperparameter changes in the mini-gpt example, to fit the generation time within the 3600s limit with my machine.
```
batch_size: 32 -> 128
max_seq_len: 100 -> 80
epochs: 30 -> 25
```
The text quality is still comparable to the current version.

**Comment by apoorvnandan** [comment]
@googlebot I signed it!

**Review by fchollet — APPROVED** [review]
LGTM, thanks a lot! 👍
